src/confd/yang/infix-if-bridge@2024-03-06.yang

```diff
@@ -297,20 +314,42 @@ submodule infix-if-bridge {
           leaf-list untagged {
             type if:interface-ref;
             description "The set of ports in the untagged set for VLAN.";
+            must "/if:interfaces/if:interface[if:name = current()]/bridge-port/bridge = ../../../../if:name" {
+              error-message "Port not a member of this bridge.";
+            }
           }
 
           leaf-list tagged {
+            description "The set of ports in the tagged set for VLAN.";
             type if:interface-ref;
-            must "not(../untagged[contains(., current())])" {
-              error-message "Cannot be an untagged port at the same time.";
+            must "(/if:interfaces/if:interface[if:name = current()]/bridge-port/bridge = ../../../../if:name
+                   or current() = ../../../../if:name) and not(../untagged[contains(., current())])" {
+              error-message "Port not member of bridge or already set as untagged.";
             }
-            description "The set of ports in the tagged set for VLAN.";
           }
         }
       }
     }
   }
 
+  augment "/if:interfaces/if:interface/ip:ipv4" {
+    leaf enabled {
+      type boolean;
```

Comment:
Do we know if the `default` property from `ietf-ip` survives the augment?

Reply:
How do you do this, do you have like an intuition for these things?!  Turns out the `default` does **not survive** our augment.  I'm gonna call you :eagle: _Eagle Eye_ from now on :astonished:

Reply:
I spent a good chunk of time yesterday trying to wrap my head around this one (too).  

From our initial testing we know that we cannot just augment with a must expression, but not only does this other augment replace the ipv4/ipv6 enabled nodes with `infix-if:enabled`, which we don't want, but it also breaks default handling as you pointed out:

 - Omitting defaults does not make the new enabled nodes inherit from original node
 - Setting defaults in the model does not take, i.e., setting IPv6 infix:enabled True

Oh, and I did I mention it also does not work?  I tested back and forth various expressions but I think libyang gets confused with multiple `enabled` nodes.  So the validation doesn't run -- even after I fixed the above issues with inference.

My take away from all is that the only safe approach here is to handle this in the callback.  We could simply just log that we skipped setting any IP address since the bridge is in VLAN filtering mode.

Reply:
After discussions with the team today they reminded me of a little thing called "deviations", which is probably what we should try to use instead.

I'm marking this PR as draft again to finish it later.